go/worker/compute/txnscheduler/service.go

```diff
@@ -15,6 +17,18 @@ func (w *Worker) SubmitTx(ctx context.Context, rq *api.SubmitTxRequest) (*api.Su
 		return nil, api.ErrUnknownRuntime
 	}
 
+	// TODO: Is this the right way to fetch current epoch?
+	epoch, err := w.commonWorker.Consensus.EpochTime().GetEpoch(ctx, consensus.HeightLatest)
```

Comment:
No this is not the right way as the consensus may have already processed the epoch transition but the worker has not.

First, move the check into the per-runtime `QueueCall` as this will give you access to the correct `committee.Node` instance (these are per-runtime as different runtime workers may have processed different epochs).

Then use the `n.commonNode.Group.GetEpochSnapshot()` to get the current epoch. Note that the epoch number is currently not exposed, but adding it should be trivial.